test/new-e2e/tests/installer/windows/install_agent_package_test.go

```diff
@@ -0,0 +1,109 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package installerwindows
+
+import (
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments/aws/host/windows"
+	"github.com/DataDog/datadog-agent/test/new-e2e/tests/installer"
+	"testing"
+)
+
+type testAgentInstallSuite struct {
+	baseSuite
+}
+
+// TestAgentInstalls tests the usage of the Datadog Installer to install the Datadog Agent package.
+func TestAgentInstalls(t *testing.T) {
+	e2e.Run(t, &testAgentInstallSuite{},
+		e2e.WithProvisioner(
+			winawshost.ProvisionerNoAgentNoFakeIntake(
+				winawshost.WithInstaller(),
+			)))
+}
+
+// TestInstallAgentPackage tests installing and uninstalling the Datadog Agent using the Datadog Installer.
+func (s *testAgentInstallSuite) TestInstallAgentPackage() {
+	s.Run("Install", func() {
+		// Arrange
+
+		// Act
+		output, err := s.installer.InstallPackage(AgentPackage)
+
+		// Assert
+		s.Require().NoErrorf(err, "failed to install the Datadog Agent package: %s", output)
+		s.Require().Host(s.Env().RemoteHost).HasARunningDatadogAgentService()
+	})
+
+	s.Run("Uninstall", func() {
```

Comment:
What happens if the install fails? Shouldn't we have a defer uninstall for safety?